src/bp/core/services/realtime/index.ts

```diff
@@ -54,11 +57,40 @@ export default class RealtimeService {
     return (eventName as string).startsWith('guest.')
   }
 
+  private makeVisitorRoomId(visitorId: string): string {
+    return `visitor:${visitorId}`
+  }
+
+  private unmakeVisitorId(roomId: string): string {
+    return roomId.split(':')[1]
+  }
+
   sendToSocket(payload: RealTimePayload) {
     debug('Send %o', payload)
     this.ee.emit(payload.eventName, payload.payload, 'server')
   }
 
+  async getVisitorIdFromSocketId(socketId: string): Promise<undefined | string> {
+    const socket = this.guest?.sockets[socketId]
+    if (!socket) {
+      return
+    }
+
+    let rooms: string[]
+    if (this.useRedis) {
+      const adapter = socket.adapter as RedisAdapter
+      rooms = await Promise.fromCallback(cb => adapter.clientRooms(socketId, cb))
+    } else {
+      rooms = Object.keys(socket.adapter.sids[socketId] ?? {})
+    }
```

Comment:
Hmm, you check locally for the socket ID first, this will not work if the user's request ends on a different server

Reply:
Yes it'll work check line #82